pkg/agentfs/docker.go

```diff
@@ -51,94 +54,249 @@ func CreateDockerfile(dir string, settingsMap map[string]string) error {
 			suffix = ".js"
 		}
 
-		// Collect all matching files
-		var options []string
-		for _, entry := range entries {
-			if strings.HasSuffix(entry.Name(), suffix) {
-				options = append(options, entry.Name())
+		// Smart entry point discovery with prioritization
+		var candidates []string
+		var priorityOrder []string
+
+		if projectType.IsPython() {
+			// Common Python entry point patterns in order of preference
+			priorityOrder = []string{
+				"main.py",         // Most common
+				"src/main.py",     // Modern src layout
+				"agent.py",        // LiveKit agents
+				"src/agent.py",    // LiveKit agents in src
+				"app.py",          // Flask/web apps
+				"src/app.py",      // Flask/web apps in src
+				"__main__.py",     // Python module entry
+				"src/__main__.py", // Python module entry in src
+			}
+		} else if projectType == ProjectTypeNode {
+			priorityOrder = []string{
+				"index.js",
+				"main.js",
+				"app.js",
+				"src/index.js",
+				"src/main.js",
+				"src/app.js",
+			}
+		}
+
+		// First, check priority patterns that exist
+		for _, pattern := range priorityOrder {
+			if _, exists := fileList[pattern]; exists {
+				candidates = append(candidates, pattern)
+			}
+		}
+
+		// Then add any other matching files not already in candidates
+		for fileName := range fileList {
+			if strings.HasSuffix(fileName, suffix) {
+				// Skip if already in candidates
+				if !slices.Contains(candidates, fileName) {
+					candidates = append(candidates, fileName)
+				}
 			}
 		}
 
+		// If we have a clear top choice, suggest it
+		if len(candidates) == 1 {
+			return candidates[0], nil
+		}
+
+		options := candidates
+
 		// If no matching files found, return early
 		if len(options) == 0 {
 			return "", nil
 		}
 
+		// Create enhanced options with descriptions
+		var selectOptions []huh.Option[string]
+		for _, option := range options {
+			var description string
+			switch {
+			case strings.Contains(option, "src/"):
+				description = fmt.Sprintf("%s (modern src/ layout)", option)
+			case strings.Contains(option, "main."):
+				description = fmt.Sprintf("%s (common main entry point)", option)
+			case strings.Contains(option, "agent."):
+				description = fmt.Sprintf("%s (LiveKit agent)", option)
+			case strings.Contains(option, "app."):
+				description = fmt.Sprintf("%s (application entry point)", option)
+			case strings.Contains(option, "__main__.py"):
+				description = fmt.Sprintf("%s (Python module entry)", option)
+			default:
+				description = option
+			}
+
+			// For Python module paths, suggest both file and module syntax
+			if projectType.IsPython() && strings.HasPrefix(option, "src/") && strings.HasSuffix(option, ".py") {
+				modulePath := strings.TrimPrefix(option, "src/")
+				modulePath = strings.TrimSuffix(modulePath, ".py")
+				modulePath = strings.ReplaceAll(modulePath, "/", ".")
+				if modulePath != "" {
+					description += fmt.Sprintf(" [can also use: src.%s]", modulePath)
+				}
+			}
```

Comment:
Not sure I understand what this is doing? What can you use e.g. `"src.agent.main"` for?